borsh/src/de/mod.rs

```diff
@@ -894,17 +912,24 @@ pub fn from_slice<T: BorshDeserialize>(v: &[u8]) -> Result<T> {
 /// Deserializes an object from a reader.
 /// # Example
 /// ```
-/// use borsh::{BorshDeserialize, BorshSerialize, from_reader};
+/// #[cfg(feature = "derive")]
+/// mod example {
+///     use borsh::{BorshDeserialize, BorshSerialize};
+///     #[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
+///     pub struct MyStruct {
+///         pub a: u64,
+///         pub b: Vec<u8>,
+///     }
+/// }
 ///
-/// #[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
-/// struct MyStruct {
-///   a: u64,
-///  b: Vec<u8>,
+/// #[cfg(feature = "derive")]
+/// {
+///     use borsh::{BorshSerialize, from_reader};
+///     let original = example::MyStruct { a: 10, b: vec![1, 2, 3] };
+///     let encoded = original.try_to_vec().unwrap();
+///     let decoded = from_reader::<_, example::MyStruct>(&mut encoded.as_slice()).unwrap();
+///     assert_eq!(original, decoded);
```

Comment:
I feel it is going to be displayed really odd on docs.rs. Let's hide irrelevant parts from the example: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example and mention that the derives are available with the `derive` feature.